timbermill-client/src/main/java/com/datorama/timbermill/pipe/TimbermillServerOutputPipe.java

```diff
@@ -0,0 +1,170 @@
+                            LOG.debug("{} were sent to Timbermill server", eventsToSend.size());
+                        }
+                    }
+                } catch (Exception e) {
+                    LOG.error("Error sending events to Timbermill server" ,e);
+                }
+            } while (buffer.size() > 0);
+        });
+        thread.start();
+    }
+
+    private void sendEventsOverConnection(HttpURLConnection httpCon, byte[] eventsWrapperBytes) throws IOException {
+        OutputStream os = httpCon.getOutputStream();
+        os.write(eventsWrapperBytes);
+        os.flush();
+        os.close();
+    }
+
+    private byte[] getEventsWrapperBytes(EventsWrapper eventsWrapper) throws JsonProcessingException {
+        ObjectMapper om = new ObjectMapper();
+        ObjectWriter ow = om.writerFor(om.getTypeFactory().constructType(EventsWrapper.class));
+        String eventsWrapperString = ow.writeValueAsString(eventsWrapper);
+        return eventsWrapperString.getBytes();
+    }
+
+    private HttpURLConnection getHttpURLConnection() throws IOException {
+        HttpURLConnection httpCon = (HttpURLConnection) timbermillServerUrl.openConnection();
+        httpCon.setRequestMethod("POST");
+        httpCon.setRequestProperty("content-type", "application/json");
+        httpCon.setDoOutput(true);
+        return httpCon;
+    }
+
+    private List<Event> createEventsToSend() throws JsonProcessingException {
+        long startBatchTime = System.currentTimeMillis();
+        List<Event> eventsToSend = new ArrayList<>();
+        int currentBatchSize = 0;
+        if(!buffer.isEmpty()) {
+            Event event = buffer.remove(0);
+            currentBatchSize += getEventSize(event);
+            eventsToSend.add(event);
+
+            while(!buffer.isEmpty() && currentBatchSize <= this.maxEventsBatchSize && !isExceededMaxTimeToWait(startBatchTime)) {
+                int eventSize = getEventSize(buffer.get(0));
+                if(currentBatchSize + eventSize <= this.maxEventsBatchSize) {
+                    event = buffer.remove(0);
+                    eventsToSend.add(event);
+                    currentBatchSize += eventSize;
+                }
+                else {
+                    break;
+                }
+            }
+        }
+        else {
+            return null;
+        }
+        return eventsToSend;
+    }
+
+    private boolean isExceededMaxTimeToWait(long startBatchTime) {
+        return System.currentTimeMillis() - startBatchTime > maxSecondsBeforeBatchTimeout * 1000;
+    }
+
+    private int getEventSize(Event event) throws JsonProcessingException {
+        ObjectMapper om = new ObjectMapper();
+        String eventStr = om.writeValueAsString(event);
+        return eventStr.getBytes().length;
+    }
+
+    @Override
+    public void send(Event e) {
+        synchronized(this) {
+            this.buffer.add(e);
+            if (this.thread == null || !this.thread.isAlive()) {
```

Comment:
let's change so that the thread will be initialized at construction same as the localOutputPipe